applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+                // Elastic Stiffness matrix; Solution Vector; RHS Vector
+                pBuilderAndSolver->ResizeAndInitializeVectors(
+                    pScheme, pStiffnessMatrix, pDx, pRHS, rModelPart);
+                // Previous Stiffness Matrix 
+                pBuilderAndSolver->ResizeAndInitializeVectors(
+                    pScheme, pStiffnessMatrixPrevious, _pDx, _pb, rModelPart);
+
+                // Mass Matrix 
+                // pBuilderAndSolver->ResizeAndInitializeVectors(
+                //     pScheme, pMassMatrix, _pDx, _pb, rModelPart);
+
+                KRATOS_INFO_IF("System Matrix Resize Time", BaseType::GetEchoLevel() > 0 && rank == 0)
+                    << system_matrix_resize_time.ElapsedSeconds() << std::endl;
+
+            }
+
+            KRATOS_INFO_IF("System Construction Time", BaseType::GetEchoLevel() > 0 && rank == 0)
+                << system_construction_time.ElapsedSeconds() << std::endl;
+            
+            // Initial operations ... things that are constant over the solution
+            // step
+            pBuilderAndSolver->InitializeSolutionStep(BaseType::GetModelPart(),
+                                                rStiffnessMatrix, rDx, rRHS);
+
+            // Initial operations ... things that are constant over the solution
+            // step
+            pScheme->InitializeSolutionStep(BaseType::GetModelPart(), rStiffnessMatrix, rDx, rRHS);
+
+            // Initialisation of the convergence criteria
+            // if (pConvergenceCriteria->GetActualizeRHSflag() == true)
+            // {
+            //     TSparseSpace::SetToZero(rRHS);
+            //     pBuilderAndSolver->BuildRHS(pScheme, rModelPart, rRHS);
+            // }
+
+            pConvergenceCriteria->InitializeSolutionStep(BaseType::GetModelPart(), pBuilderAndSolver->GetDofSet(), rStiffnessMatrix, rDx, rRHS);
+
+            // if (pConvergenceCriteria->GetActualizeRHSflag() == true)
+            //     TSparseSpace::SetToZero(rRHS);
+
+            mSolutionStepIsInitialized = true;
+            KRATOS_INFO_IF("PrebucklingStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+                << "Exiting InitializeSolutionStep" << std::endl;
+        }
+        KRATOS_CATCH("")
+    }
+
+    bool SolveSolutionStep() override
+    {
+        KRATOS_TRY
+
+        ModelPart &rModelPart = BaseType::GetModelPart();
+        ProcessInfo& rProcessInfo = rModelPart.GetProcessInfo();
+
+        const int rank = rModelPart.GetCommunicator().MyPID();
+
+        SchemePointerType &pScheme = this->pGetScheme();
+        BuilderAndSolverPointerType &pBuilderAndSolver = this->pGetBuilderAndSolver();
+        //SparseMatrixType &rMassMatrix = this->GetMassMatrix();
+        SparseMatrixType &rStiffnessMatrix = this->GetStiffnessMatrix();
+        SparseMatrixType &rStiffnessMatrixPrevious = this->GetStiffnessMatrixPrevious();
+        SparseVectorType &rDx = this->GetDx();
+        SparseVectorType &rRHS = this->GetRHS();
```

Comment:
why don't you just access the member pointer directly? Would be much cleaner IMO